docs/aws/audit/sgaudit/rules/ec2_public_instances_exist_remediation.mdx

```diff
@@ -1,6 +1,221 @@
+
+3. **Edit Inbound Rules:**
+   - With the Security Group selected, go to the "Inbound rules" tab.
+   - Click on the "Edit inbound rules" button.
+
+4. **Remove or Restrict Public Access:**
+   - Review the existing inbound rules. Look for rules that allow access from `0.0.0.0/0` (which means any IP address can access the instance).
+   - Remove these rules or modify them to restrict access to specific IP addresses or ranges that you trust.
+   - Click "Save rules" to apply the changes.
+
+By following these steps, you can ensure that your EC2 instances are not publicly accessible, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent EC2 instances from being publicly accessible in Security Groups using AWS CLI, you can follow these steps:
+
+1. **Identify the Security Group:**
+   First, identify the security group associated with your EC2 instance. You can list all security groups and their details using the following command:
+   ```sh
+   aws ec2 describe-security-groups
+   ```
+
+2. **Revoke Inbound Rules Allowing Public Access:**
+   Revoke any inbound rules that allow public access (i.e., rules with a source of `0.0.0.0/0` for IPv4 or `::/0` for IPv6). For example, to revoke an inbound rule that allows SSH (port 22) from any IP address, use:
+   ```sh
+   aws ec2 revoke-security-group-ingress --group-id <security-group-id> --protocol tcp --port 22 --cidr 0.0.0.0/0
+   ```
+
+3. **Add Restrictive Inbound Rules:**
+   Add more restrictive inbound rules to allow access only from specific IP addresses or CIDR blocks. For example, to allow SSH access only from a specific IP address (e.g., `203.0.113.0`), use:
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-id <security-group-id> --protocol tcp --port 22 --cidr 203.0.113.0/32
+   ```
+
+4. **Verify Security Group Rules:**
+   Verify that the security group rules have been updated correctly and no rules allow public access. You can describe the security group again to check the current rules:
+   ```sh
+   aws ec2 describe-security-groups --group-ids <security-group-id>
+   ```
+
+By following these steps, you can ensure that your EC2 instances are not publicly accessible through their associated security groups.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent EC2 instances from being publicly accessible in Security Groups using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session and create an EC2 client to interact with the AWS EC2 service.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>